### PR TITLE
KOGITO-3224 - Enable explainability messaging in trusty image

### PR DESCRIPTION
**Pull request by r00ta**
jira: https://issues.redhat.com/browse/KOGITO-3224

The explainability messaging has to be enabled in the trusty image. 

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by ricardozanini** [comment]
@spolti mind taking a look?

**Review by ricardozanini — APPROVED** [review]

**Comment by danielezonca** [comment]
@sutaakar 
Can you please have a look to this PR too?

File: modules/kogito-explainability/added/kogito-app-launch.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ CONFIGURE_SCRIPTS=(
 source ${KOGITO_HOME}/launch/configure.sh
 #############################################
 
-if [ ${EXPLAINABILITY_COMMUNICATION^^} = "REST" ] ; then 
+if [ "${EXPLAINABILITY_COMMUNICATION^^}" = "REST" ] ; then
```
Comment (spolti) [inline modules/kogito-explainability/added/kogito-app-launch.sh]: don't do this on the launch-script. use this PR as example:  https://github.com/kiegroup/kogito-images/pull/230

**Review by spolti — CHANGES_REQUESTED** [review]

File: modules/kogito-explainability/added/kogito-app-launch.sh (hunk 1 above)
Comment (r00ta) [reply modules/kogito-explainability/added/kogito-app-launch.sh]: Hi @spolti , I refactored it. Let me know!

File: modules/kogito-explainability/added/launch/kogito-explainability.sh [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+    configure_explainability_jar
 }
 
 function configure_explainability_http_port {
     local httpPort=${HTTP_PORT:-8080}
     KOGITO_EXPLAINABILITY_PROPS="${KOGITO_EXPLAINABILITY_PROPS} -Dquarkus.http.port=${httpPort}"
 }
 
+function configure_explainability_jar {
+    local allowed_communication_types=("REST", "MESSAGING")
```
Comment (spolti) [inline modules/kogito-explainability/added/launch/kogito-explainability.sh]: no need to have comma here.

File: modules/kogito-trusty/added/launch/kogito-trusty.sh [hunk 3]
```diff
@@ -3,15 +3,26 @@
...
+    enable_explainability
 }
 
 function configure_trusty_http_port {
     local httpPort=${HTTP_PORT:-8080}
     KOGITO_TRUSTY_PROPS="${KOGITO_TRUSTY_PROPS} -Dquarkus.http.port=${httpPort}"
 }
 
+function enable_explainability {
+    local explainabilityEnabled=${EXPLAINABILITY_ENABLED^^}
```
Comment (spolti) [inline modules/kogito-trusty/added/launch/kogito-trusty.sh]: why not have something like this:
```
local explainabilityEnabled=${EXPLAINABILITY_ENABLED:"false"}
if [ "${explainabilityEnabled^^}" = "TRUE" ] ; then 
      explainabilityEnabled="true"
fi
KOGITO_TRUSTY_PROPS="${KOGITO_TRUSTY_PROPS} -Dtrusty.explainability.enabled=${explainabilityEnabled}"
```

the default value is just a example.

**Review by spolti — CHANGES_REQUESTED** [review]

File: modules/kogito-trusty/added/launch/kogito-trusty.sh (hunk 3 above)
Comment (r00ta) [reply modules/kogito-trusty/added/launch/kogito-trusty.sh]: Done

File: modules/kogito-trusty/added/launch/kogito-trusty.sh [hunk 4]
```diff
@@ -18,11 +18,9 @@ function configure_trusty_http_port {
 }
 
 function enable_explainability {
-    local explainabilityEnabled=${EXPLAINABILITY_ENABLED^^}
-    if [ "${explainabilityEnabled}" = "FALSE" ] ; then 
-      explainabilityEnabled="false"
-    else
-      explainabilityEnabled="true"
+    local explainabilityEnabled=${EXPLAINABILITY_ENABLED:"true"}
+    if [ "${explainabilityEnabled^^}" = "TRUE" ] ; then
```
Comment (spolti) [inline modules/kogito-trusty/added/launch/kogito-trusty.sh]: if the default value is true, shouldn't this if test if it is false?

File: modules/kogito-trusty/added/launch/kogito-trusty.sh [hunk 5]
```diff
@@ -3,15 +3,24 @@
...
+    enable_explainability
 }
 
 function configure_trusty_http_port {
     local httpPort=${HTTP_PORT:-8080}
     KOGITO_TRUSTY_PROPS="${KOGITO_TRUSTY_PROPS} -Dquarkus.http.port=${httpPort}"
 }
 
+function enable_explainability {
+    local explainabilityEnabled=${EXPLAINABILITY_ENABLED:="true"}
```
Comment (spolti) [inline modules/kogito-trusty/added/launch/kogito-trusty.sh]: what this "=" means here?
it should be only ":"

File: modules/kogito-trusty/added/launch/kogito-trusty.sh [hunk 6]
```diff
@@ -3,15 +3,24 @@
...
+    if [ "${explainabilityEnabled^^}" = "FALSE" ] ; then
```
Comment (spolti) [inline modules/kogito-trusty/added/launch/kogito-trusty.sh]: what happens if user set a nonsense value for this env? Maybe worth to also treat it and default to true if that happens.

**Review by spolti — CHANGES_REQUESTED** [review]

File: modules/kogito-trusty/added/launch/kogito-trusty.sh (hunk 4 above)
Comment (r00ta) [reply modules/kogito-trusty/added/launch/kogito-trusty.sh]: Yep, I changed it immediately after the commit

File: modules/kogito-trusty/added/launch/kogito-trusty.sh (hunk 5 above)
Comment (r00ta) [reply modules/kogito-trusty/added/launch/kogito-trusty.sh]: Using just `=` the default value is not assigned right? https://wiki.bash-hackers.org/syntax/pe#assign_a_default_value

File: modules/kogito-trusty/added/launch/kogito-trusty.sh (hunk 6 above)
Comment (r00ta) [reply modules/kogito-trusty/added/launch/kogito-trusty.sh]: done

File: modules/kogito-trusty/added/launch/kogito-trusty.sh (hunk 5 above)
Comment (r00ta) [reply modules/kogito-trusty/added/launch/kogito-trusty.sh]: btw I've refactored this function and this instruction is not used anymore

**Comment by r00ta** [comment]
Hi @spolti , I've refactored the pr according to your comments, let me know if there is anything else

**Review by Kaitou786 — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]
Looks good to me.

**Comment by spolti** [comment]
@r00ta sounds good now, let's wait CI finish, the failing tests were fixed.

**Review by spolti — APPROVED** [review]

**Pull request merged**